main/src/library/File.flix

```diff
@@ -800,4 +800,19 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Creates the directory `d`.
+    ///
+    /// Returns `true` if the directory `d` was created and did not already exist.
+    ///
+    pub def mkdir(d: String): Result[Bool, String] = ???
+
+    ///
+    /// Creates the directory `d` along with all necessary parent directories.
+    ///
+    /// Returns `true` if the directory `d` was created and did not already exist.
```

Comment:
Is this okay?
I split the description up to make it shorter. It is more like the `def write*`'s now.